.github/workflows/buildJDK.yml

```
@@ -101,6 +101,17 @@ jobs:
         ${MANDREL_HOME}/bin/native-image HelloStrict
         ./hellostrict | tee native.txt
         diff java.txt native.txt
+        echo "
+        import org.graalvm.home.Version;
+        public class VersionTest {
+            public static void main(String[] args) {
+                System.out.println(Version.getCurrent());
+            }
+        }
+        " > VersionTest.java
+        ${MANDREL_HOME}/bin/javac -cp ${MANDREL_HOME}/lib/jvmci/graal-sdk.jar VersionTest.java
+        ${MANDREL_HOME}/bin/native-image VersionTest
+        ./versiontest | grep ${MANDREL_VERSION}
```

I suggest extending it to test both build-time and run-time values:

```suggestion
        echo "
        import org.graalvm.home.Version;
        public class VersionTest {
            final static String version = Version.getCurrent().toString();
            public static void main(String[] args) {
                System.out.println("Baked in: " + version);
                System.out.println("At runtime: " + Version.getCurrent());
            }
        }
        " > VersionTest.java
        ${MANDREL_HOME}/bin/javac -cp ${MANDREL_HOME}/lib/jvmci/graal-sdk.jar VersionTest.java
        ${MANDREL_HOME}/bin/native-image --intialize-at-build-time=. VersionTest
        (($(./versiontest | grep ${MANDREL_VERSION} | wc -l) == 2))
```

---

Yes, agreed. Though, I'm thinking we mostly care about the `at image build time` value (i.e. "Baked in" in your example). That's the problem we ran into in https://github.com/Karm/mandrel-integration-tests/pull/144

---

Done in the update. I don't think we necessarily care about the runtime version, do we?